packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -50,7 +50,9 @@ export interface TemplateCache {
 }
 
 export interface SlotSet {
-    [key: string]: VNodes;
+    // Slot assignments by name
+    slotAssignments: { [key: string]: VNodes };
+    owner?: VM;
```

Comment:
Why isn't the `owner` property required? All the VNodes associated with the `slotAssignments` has to have an owner.

Reply:
The root vm will not have an owner for its slotset(its `slotAssignments` are also empty). One option is to make `VM['cmpSlots']` an optional field. It introduces a few checks every time `vm.cmpSlots` are accessed. I felt this was a better tradeoff.

Reply:
> One option is to make `VM['cmpSlots']` an optional field.

I think this is a preferable approach to avoid allocating 2 empty objects for all the components that don't accept slots.